### PR TITLE
fix: flickering on highlights

### PR DESCRIPTION
**Pull request by 0tuedon**

### CODE REVIEW

**Comment by 0tuedon** [comment]
#124

File: apps/extension/src/hooks/useNostrHighlights.tsx [hunk 1]
```diff
@@ -20,7 +20,13 @@ export const useNostrHighlights = () => {
         .then((outcome: ActionResponse) => {
           if (outcome?.success) {
             const data = (outcome as SucccessAcionResponse).data;
-            setHighlights(data as IHighlight[]);
+            // Check if every highlight has a text property before setting
+            const isAllDataWithText = data.every(highlight => highlight?.text)
+            if(isAllDataWithText && data.length > 0 ){
+             setHighlights(data as IHighlight[]);
+            }
+            else{
```
Comment (Extheoisah) [inline apps/extension/src/hooks/useNostrHighlights.tsx:28]: else what??

File: apps/extension/src/hooks/useNostrHighlights.tsx [hunk 2]
```diff
@@ -29,7 +35,7 @@ export const useNostrHighlights = () => {
     };
 
     pollHighlights();
-    const timer = setInterval(pollHighlights, 1000);
+    const timer = setInterval(pollHighlights, 3000);
```
Comment (Extheoisah) [inline apps/extension/src/hooks/useNostrHighlights.tsx:38]: why increase the time?

**Review by Extheoisah — CHANGES_REQUESTED** [review]
A few comments but I think the underlying question is why are there highlights without texts?
We can merge this as a temp fix though.

File: apps/extension/src/hooks/useNostrHighlights.tsx (hunk 1 above)
Comment (0tuedon) [reply apps/extension/src/hooks/useNostrHighlights.tsx]: added a breakpoint

File: apps/extension/src/hooks/useNostrHighlights.tsx (hunk 2 above)
Comment (0tuedon) [reply apps/extension/src/hooks/useNostrHighlights.tsx]: was for testing purposes, reverted

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**